src/app/components/page-header.tsx

```diff
@@ -0,0 +1,85 @@
+import { memo } from 'react';
+import { FiArrowLeft, FiX } from 'react-icons/fi';
+import { useNavigate } from 'react-router-dom';
+
+import { Box, Flex, FlexProps, IconButton } from '@stacks/ui';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { NetworkModeBadge } from '@app/components/network-mode-badge';
+import { Title } from '@app/components/typography';
+
+interface PageHeaderProps extends FlexProps {
+  actionButton?: JSX.Element;
+  closeIcon?: boolean;
+  hideActions?: boolean;
+  onClose?(): void;
+  enableClose?: boolean;
+  enableGoBack?: boolean;
+  title?: string;
+}
+export const PageHeader: React.FC<PageHeaderProps> = memo(props => {
+  const {
+    actionButton,
+    hideActions,
+    onClose,
+    closeIcon,
+    title,
+    enableGoBack,
+    enableClose,
+    ...rest
+  } = props;
+  const navigate = useNavigate();
+
+  function defaultCloseAction() {
+    navigate(RouteUrls.Home);
+  }
+  return (
+    <Flex
+      alignItems={hideActions ? 'center' : 'flex-start'}
+      justifyContent="space-between"
+      p="base"
+      position="relative"
+      {...rest}
+    >
+      {enableGoBack ? (
+        <Box flexBasis="20%" onClick={() => navigate('..')} as="button">
+          <IconButton alignSelf="center" icon={FiArrowLeft} iconSize="24px" />
+        </Box>
+      ) : (
+        <Box flexBasis="20%" />
+      )}
+
+      <Flex alignItems="center" flex="60%" justifyContent="center">
+        <Title
+          alignSelf="center"
+          fontSize="16px"
+          fontWeight={500}
+          lineHeight="24px"
+          textAlign="center"
+          {...rest}
+        >
+          {title}
+        </Title>
```

Comment:
`title` is described as a prop as optional, but we always render it.

Either it should be a required prop, or we render the title conditionally.